notebooks/03-query/15-pivot-unpivot.ipynb

```diff
@@ -0,0 +1,372 @@
+    "       COALESCE(\"human_added\",0) AS \"human_added\", \n",
+    "       COALESCE(\"robot_deleted\",0) AS \"robot_deleted\", \n",
+    "       COALESCE(\"human_deleted\",0) AS \"human_deleted\"\n",
+    "    FROM \"example-pivot-unpivot\" \n",
+    "    \n",
+    "    PIVOT ( SUM(added) as added, \n",
+    "            SUM(deleted) as deleted \n",
+    "            \n",
+    "            FOR \"isRobot\" IN ('true' as robot, 'false' as human)\n",
+    "          )\n",
+    ")x\n",
+    "GROUP BY 1\n",
+    "ORDER BY 6\n",
+    "'''\n",
+    "display_client.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3f0af833-9508-4f26-a484-2778ca857a2a",
+   "metadata": {},
+   "source": [
+    "## UNPIVOT example\n",
+    "Given the prior query we can now investigate the sources of the updates in the most robotically updated pages.\n",
+    "```\n",
+    "  UNPIVOT ( \"changes\"  FOR \"action\" IN (\"added\", \"deleted\") )\n",
+    "```\n",
+    "This example uses UNPIVOT to:\n",
+    "- take the values of multiple columns: `\"added\",\"deleted\"`\n",
+    "- incorporate them into a single column called the \"values column\": `\"changes\"`\n",
+    "- while expanding the result with a \"names column\" containing the names of the pivoted columns: `\"action\"`\n",
+    "\n",
+    "The columns names become row values.\n",
+    "\n",
+    "In this example we examine the most active user as measured by total adding or deleting of text in the wikipedia pages in the channel `'#it.wikipedia'` which had a ratio of about 0.50 (half robots):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f0159df8-6ca0-45af-9c68-6acd92defc4f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT \"user\",\n",
+    "       \"action\",\n",
+    "       SUM(\"changes\") \"total_changes\"\n",
+    "FROM \"example-pivot-unpivot\"\n",
+    "\n",
+    "UNPIVOT ( \"changes\"  FOR \"action\" IN (\"added\", \"deleted\") )\n",
+    "\n",
+    "WHERE \"channel\"='#it.wikipedia' AND \"isRobot\"='true'\n",
+    "GROUP BY 1,2\n",
+    "ORDER BY 3 DESC \n",
+    "'''\n",
+    "display_client.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "44738d6d-cec2-40ad-aaba-998c758c63f4",
+   "metadata": {},
+   "source": [
+    "## Clean up\n",
+    "\n",
+    "Run the following cell to remove everything used in this notebook from the database and data generation engine."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8082b545-ba7f-4ede-bb6e-2a6dd62ba0d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "druid.datasources.drop(\"example-pivot-unpivot\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "54b8d5fe-ba85-4b5b-9669-0dd47dfbccd1",
+   "metadata": {},
+   "source": [
+    "## Summary\n",
+    "\n",
+    "You learned about how to use the new experimental functions PIVOT and UNPIVOT."
```

Comment:
Suggestion: Instead of telling the user what they learned, summarize what PIVOT and UNPIVOT operations do in Druid.

Reply:
done